core/kazoo_modb/src/kazoo_modb_migrate_maintenance.erl

```diff
@@ -0,0 +1,367 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2017, 2600Hz INC
+%%% @doc
+%%% The Great Kazoo Migration (TM)
+%%% @end
+%%%-------------------------------------------------------------------
+-module(kazoo_modb_migrate_maintenance).
+
+-export([migrate/0
+        ,migrate_account/1
+
+        ,migrate_voicemails/1
+        ,migrate_cdrs/1
+        ]).
+
+-include("kazoo_modb.hrl").
+
+-type update_funs() :: [fun((kz_json:object()) -> kz_json:object())].
+
+-spec migrate() -> 'ok'.
+migrate() ->
+    AccountIds = kz_term:shuffle_list(kapps_util:get_all_accounts('raw')),
+    Total = length(AccountIds),
+    io:format("Start migrating items to MODB from ~b databases~n~n", [Total]),
+    lists:foldl(fun(A, C) -> migrate_account_fold(A, C, Total) end, 1, AccountIds),
+    'ok'.
+
+-spec migrate_account_fold(ne_binary(), non_neg_integer(), non_neg_integer()) -> non_neg_integer().
+migrate_account_fold(AccountId, Current, Total) ->
+    io:format("(~p/~p) migrating items within account '~s'~n", [Current, Total, AccountId]),
+    _ = migrate_account(AccountId),
+    io:format("[~s] finished migrating items~n~n", [AccountId]),
+    Current + 1.
+
+-spec migrate_account(ne_binary()) -> 'ok'.
+migrate_account(Account) ->
+    lists:foreach(fun(Fun) -> Fun(Account) end
+                 ,[fun migrate_voicemails/1
+                  ,fun migrate_cdrs/1
+                  ]
+                 ).
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+-spec get_view_count(ne_binary(), ne_binary()) -> non_neg_integer().
+get_view_count(AccountId, View) ->
+    get_view_count(AccountId, View, 2).
+
+-spec get_view_count(ne_binary(), ne_binary(), integer()) -> non_neg_integer().
+get_view_count(_AccountId, _View, Retry) when Retry < 0 ->
+    io:format("[~s] failed to fetch view ~s count~n", [_AccountId, _View]),
+    0;
+get_view_count(AccountId, View, Retry) ->
+    AccountDb = kz_util:format_account_db(AccountId),
+    case kz_datamgr:get_results_count(AccountDb, View, []) of
+        {'ok', Total} -> Total;
+        {'error', 'not_found'} ->
+            kapps_maintenance:refresh(AccountDb),
+            get_view_count(AccountDb, View, Retry-1);
+        {'error', _} ->
+            get_view_count(AccountDb, View, Retry-1)
+    end.
+
+-spec next_skip(non_neg_integer(), non_neg_integer(), kz_proplist()) -> kz_proplist().
+next_skip(0, Skip, ViewOptions) -> props:set_value('skip', Skip, ViewOptions);
+next_skip(Remaining, Skip, ViewOptions) ->
+    case Remaining < props:get_value(limit, ViewOptions) of
+        true -> props:set_value('skip', Skip, props:delete(limit, ViewOptions));
+        false -> props:set_value('skip', Skip, ViewOptions)
+    end.
+
+-spec maps_update_with(ne_binary(), fun((any()) -> any()), any(), map()) -> map().
+maps_update_with(Key, UpdateFun, Init, Map) ->
```

Comment:
Interesting. I'm ok with this then, but we should keep this kind of things in mind since the `maps` implementation can change in the future (to a C version for example).